src/wp-includes/sitemaps/class-wp-sitemaps-index.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Sitemaps: WP_Sitemaps_Index class.
+ *
+ * Generates the sitemap index.
+ *
+ * @package WordPress
+ * @subpackage Sitemaps
+ * @since 5.5.0
+ */
+
+/**
+ * Class WP_Sitemaps_Index.
+ * Builds the sitemap index page that lists the links to all of the sitemaps.
+ *
+ * @since 5.5.0
+ */
+class WP_Sitemaps_Index {
+
+	/**
+	 * The main registry of supported sitemaps.
+	 *
+	 * @since 5.5.0
+	 * @var WP_Sitemaps_Registry
+	 */
+	protected $registry;
+
+	/**
+	 * WP_Sitemaps_Index constructor.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @param WP_Sitemaps_Registry $registry Sitemap provider registry.
+	 */
+	public function __construct( $registry ) {
```

Comment:
Type check 
```suggestion
	public function __construct( WP_Sitemaps_Registry  $registry ) {
```

Reply:
Relevant post: https://make.wordpress.org/core/2020/03/20/updating-the-coding-standards-for-modern-php/

Reply:
I think we should add them wherever possible (classes), they'd only have positive effects.

Reply:
The post above says „ Using the array keyword in type declarations is strongly discouraged for now„

Reply:
Well then let's just do it for classes. There's a precedence in core, and again no harm.